app/views/shared/_select_work_type_modal.html.erb

```diff
@@ -7,12 +8,14 @@
       </div>
       <form class="new-work-select">
         <div class="modal-body">
+          <% collection_id = params[:add_works_to_collection] %>
+          <% collection_id ||= params['id'] if params['id'] && params['controller'] == 'hyrax/dashboard/collections' %>
```

Comment:
I would prefer to see this handled in the presenter if possible, rather than having this logic in a view partial. Now that the presenter branches on `collection_id`, you could define a new method in the presenter to handle this case?

Reply:
Moved to presenter.